### PR TITLE
test(authorization): front-run address(0) authorization

### PR DESCRIPTION
**Pull request by Rubilmax**

### CODE REVIEW

**Review by Jean-Grimal — APPROVED** [review]

File: test/forge/integration/AuthorizationIntegrationTest.sol [hunk 1]
```diff
@@ -127,14 +127,23 @@ contract AuthorizationIntegrationTest is BaseTest {
         authorization.nonce = 0;
         authorization.authorizer = vm.addr(privateKey);
 
+        address authorized = _boundAddressNotZero(authorization.authorized);
+        authorization.authorized = address(0);
```
Comment (MerlinEgalite) [inline test/forge/integration/AuthorizationIntegrationTest.sol:131]: We should create an independent test I think

Comment (Rubilmax) [reply test/forge/integration/AuthorizationIntegrationTest.sol:131]: But it would essentially be the same test whereas we can test both at the same time

Comment (MerlinEgalite) [reply test/forge/integration/AuthorizationIntegrationTest.sol:131]: Then can we move `address authorized = _boundAddressNotZero(authorization.authorized);` lower. Else it seems weird to query it now and set `authorization.authorized` to the address zero.

Comment (Rubilmax) [reply test/forge/integration/AuthorizationIntegrationTest.sol:152]: i pushed a split because you were right it wasn't testing the same thing

**Review by MerlinEgalite — APPROVED** [review]

**Review by QGarchery — COMMENTED** [review]
LGTM, but to adapt depending on whether we use #652 or not (see https://github.com/morpho-org/morpho-blue/pull/652#issuecomment-1864086371)

**Pull request closed without merging**